src/Symfony/Component/Console/Command/Command.php

```diff
@@ -273,12 +267,25 @@ public function run(InputInterface $input, OutputInterface $output): int
 
         $input->validate();
 
+        $lockName = $this->resolveLockName($input);
+
+        if (null !== $lockName) {
+            $lockHandler = new LockHandler();
```

Comment:
This is a static factory, we need a way to use dependency injection to use database-backed locks.

Reply:
Not really sure about the right way to do it, so I registered the `LockHandler` with the service ID `console.lock.handler`, to inject it in every commands, with the `AddConsoleCommandPass`. I'm now injecting the `lock.default.factory` inside the `LockHandler`.

Is there a better approach for it ?